src/libraries/System.Text.Json/tests/Common/ExtensionDataTests.cs

```diff
@@ -614,6 +614,28 @@ public async Task ExtensionPropertyDuplicateNames(JsonSerializerOptions options)
             Assert.Null(obj.MyOverflow);
         }
 
+        [Theory]
+        [InlineData("""{ "1": 0   , "1": 1    }""")]
+        [InlineData("""{ "1": null, "1": null }""")]
+        [InlineData("""{ "1": "a" , "1": null }""")]
+        [InlineData("""{ "1": null, "1": "b"  }""")]
```

Comment:
perhaps I missed but I think there might be couple of test cases which would be nice to see:
- recursive structure with duplicate properties in one of the inner, both positive and negative test cases
- two C# properties with same name but different casing
- are duplicates allowed inside of dictionary? regardless of the answer there should be a test case
- if there is per type setting on JsonTypeInfo I'd imagine this should be exercised (I think no but have only skimmed through PR)